### ISSUE TITLE
Update firmware in LAN-only mode, or at least without login

### ISSUE THREAD

**Issue by ccloli** [bambulab/BambuStudio#4530]
**Is your feature request related to a problem? Please describe.**

There's no firmware updates in LAN-only mode, it always says it's the latest firmware, even though any normal people would say `01.03.00.00` is higher than the "latest" `01.02.00.07`.

There's already a bunch of discussion about updating firmware in LAN-only mode, like:

- https://github.com/bambulab/BambuStudio/issues/1580
- https://github.com/bambulab/BambuStudio/issues/4026
- https://forum.bambulab.com/t/fyi-lan-only-model-no-notification-of-firmware-updates/8955
- https://forum.bambulab.com/t/remove-lan-mode-limitations/51792

If you said the LAN-only mode is not connected to the server, so the printer cannot receive OTA. But even though you've already turned off the LAN-only mode, ***if you don't login your account in printer***, you still cannot receive OTA.

![image](https://github.com/user-attachments/assets/71b99b5e-ad58-4671-be4c-7e0571a16f04)

And once you logged in the printer with an account, it shows the update in Bambu Studio.

![image](https://github.com/user-attachments/assets/2dded69b-007c-4ac2-91ce-f3d462f24de5)

**Describe the solution you'd like**

- For LAN-only mode: Allow manual download & flash firmware
  - You can do this by putting the firmware in SD card, and load it from SD card.
  - Another way is to download the firmware in Bambu Studio, then transfer the firmware file to the printer in LAN.
- For updating requires login: Allow checking updates without login
  - You've already connected to the server, because it checks whether the region is valid or not, then shows the dynamic login QR code. I don't believe this can be done without Internet.

**Describe alternatives you've considered**

N/A, it's already cleared.

**Additional context**

People DO care about privacy, not everyone likes putting everything on or through the cloud, I believe that's the reason why you introduced the LAN-only mode. So please consider improving the LAN-only mode, like the features listed in https://github.com/bambulab/BambuStudio/issues/4512 .

**Comment by digitalcohesion** [comment]
Yes please. Upgrading firmware should not require apps and accounts and everything that goes along with it. This is the primary reason I would not recommend a Bambu Labs printer to anyone who asks me about mine. Knowing what I know now I would not buy one.

**Comment by Artoria2e5** [comment]
Well now they have added the feature to P1, but you gotta do a firmware update to get it… Don’t get me wrong, I’m happy that new buyers will get to update without an account, but it sure looks like I am going to need to log in for this old 01.03.00.00 machine I have.
